hermes/src/wormhole.rs

```diff
@@ -102,11 +102,31 @@ pub async fn verify_vaa<'a>(
             )
         })?;
 
-    let mut num_correct_signers = 0;
-    for sig in header.signatures.iter() {
-        let signer_id: usize = sig.index.into();
+    let mut last_signed_id = None;
+    let mut valid_signatures = vec![];
+    for signature in header.signatures.into_iter() {
+        let signer_id: usize = signature.index.into();
+
+        if let Some(last_signed_id) = last_signed_id {
+            if last_signed_id >= signer_id {
+                return Err(anyhow!(
+                    "Signatures are not sorted by signer ID. Last signer ID: {}, current signer ID: {}",
+                    last_signed_id,
+                    signer_id
+                ));
+            }
+        }
+        last_signed_id = Some(signer_id);
```

Comment:
Nitpick: could be at the bottom of the function, feels like the correct place to go as It's after everything.